### PR TITLE
Proposed repo strategy

### PR DESCRIPTION
**Pull request by texasmichelle**
Define and describe example types (end-to-end, component-based, third-party hosted)
Define requirements for housing examples
Update list of ideas for additional examples

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/117)

### CODE REVIEW

**Comment by puneith** [comment]
@texasmichelle I also see some common steps for the examples README which could be pulled into  "common" or something like that.

**Comment by cwbeitel** [comment]
If you are to migrate the category definitions to the contrib guide, in regard to "This repository is home to three types of examples...", I don't think "third-party hosted" is a category of example that this repository hosts. It's an important category of example to explain (and good to maintain a list of these in the readme) but I was a bit confused about it at first.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,18 +1,99 @@
...
 You want to contribute to Kubeflow examples? That's awesome! Please refer to the short guide below. 
 
-# Contributing Guide
+The [Kubeflow](https://github.com/kubeflow/kubeflow/blob/master/README.md) project is dedicated to
+making machine learning on Kubernetes simple, portable, and scalable. We need your support in making
+this repo the destination for top models and examples that highlight the power of Kubeflow.
+
+## Example types
+
+This repository is home to three types of examples, described in detail in the [README](./README.md):
```
[the remark this answers is not in the document]
Comment (texasmichelle) [reply CONTRIBUTING.md]: Makes sense! Done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,18 +1,99 @@
...
+
+Improvements to existing end-to-end and component-focused examples are always welcome. This could
+mean smoothing out any rough edges, filling out support for additional runtime environments, fixing
+bugs, and improving clarity in the instructions.
+
+Additional component-focused examples for highly requested concepts are also encouraged. This could
+include techniques such as:
+* Recommendation
+* Classification
+* Vision
```
[the remark this answers is not in the document]
Comment (texasmichelle) [reply CONTRIBUTING.md]: Good point

File: README.md [hunk 3]
```diff
@@ -1,18 +1,83 @@
 # kubeflow-examples
```
[the remark this answers is not in the document]
Comment (texasmichelle) [reply README.md]: Did we ever come to a consensus on where the seldon examples should live? @cliveseldon 
There's a case to be made for moving them in here.

[the remark this answers is not in the document]
Comment (cliveseldon) [reply README.md]: Probably makes sense to have all examples in one place, so I'd be happy if https://github.com/kubeflow/example-seldon were moved inside this repo.

**Comment by texasmichelle** [comment]
@cwbeitel Thanks for the detailed review! How do these changes look?

**Comment by texasmichelle** [comment]
@ankushagarwal could I get an approval?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (texasmichelle) [reply README.md]: #123 created

**Comment by cwbeitel** [comment]
/lgtm

@texasmichelle Looks great!

I don't want to block this with extra comments if it's useful to get a first version out there but here are a few we can push to the next version if you want:

- If you're going to say "component-focused" the I would say "application-focused" not "application-specific" because the suffix has the same meaning in both cases.

- Under the component-focused heading you clarify that a component-focused example is meant to be smaller in scope than an end-to-end example and this is more or less implied about an application-focused example but might be best to state it there as well.

- Under "how to contribute" I think it would be helpful to give people a picture of what a good issue should contain. Perhaps goals, non-goals, audience, etc. is something more organizational that belongs in the issue not necessarily the example readme.

**Comment by texasmichelle** [comment]
I came here to say that those are great suggestions and any of us would be happy to review a PR with further fine-tuning, but I saw that you already have #124 open :) Thanks!!

**Pull request merged**